src/relay/WalletSDKRelay.ts

```diff
@@ -1046,14 +1046,11 @@ export class WalletSDKRelay extends WalletSDKRelayAbstract {
         })
 
         const _cancel = (error?: Error) => {
-          if (error) {
+          if (error && (error instanceof ErrorResponse)) {
             this.handleWeb3ResponseMessage(
               Web3ResponseMessage({
                 id,
-                response: ErrorResponse(
-                  Web3Method.switchEthereumChain,
-                  error.message
-                )
+                response: (error as ErrorResponse)
```

Comment:
is this typecast still needed? i'd expect TS to be able to type-narrow after the instanceof check above.

Reply:
yeah i hoped so, but for some reason it still thinks that's just `Error`